ExRam.Gremlinq.Core/Models/Metadata/MetadataStore.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace ExRam.Gremlinq.Core
+{
+    public class MetadataStore : IMetadataStore
+    {
+        private readonly ConcurrentDictionary<Type, ConcurrentDictionary<string, PropertyMetadata>> _propertyMetadataDictionary;
```

Comment:
Please keep the whole model immutable, including the metadata store. It's fine having the builders mutable though.

Reply:
I reworked the code a bit to drop the MetadataStore class and its use in the IGraphModel interface. At this point, the only mutations should occur during the processing of the Action passed to the .Configure method of the model. After that, the metadata is readonly and immutable.

I was also careful not to expose the inner workings of this to the public interfaces.